### PR TITLE
do not delete wallet files

### PR DESCRIPTION
**Pull request by peterjah**

### CODE REVIEW

**Review by Thykof — APPROVED** [review]
code complexity can be reduce by writing small functions.

I tested on macos it works as expected.

File: pkg/plugin/manager.go [hunk 1]
```diff
@@ -156,9 +156,31 @@ func (m *Manager) Delete(correlationID string) error {
...
+					return fmt.Errorf("deleting file %s: %w", file.Name(), err)
+				}
+			}
+		}
+	} else {
+		err = os.RemoveAll(pluginDir)
+		if err != nil {
+			return fmt.Errorf("deleting plugin %s: %w", correlationID, err)
+		}
 	}
```
Comment (gregLibert) [inline pkg/plugin/manager.go:184]: You could improve your code by:
- changing the order of your if and return earlier
- put all that in a removePlugin function
- unit test the removePlugin function

My suggestion:
```suggestion
	err = removePlugin(plgn)
	if err != nil {
		return fmt.Errorf("deleting plugin %s: %w", correlationID, err)
	} 
	
```

With the following function:
```golang
// removePlugin removes all the plugin files from the file system.
// However, for the "massa-labs/massa-wallet" plugin, it keeps the wallet files.
// The wallet files are identified by the prefix "wallet_" and the suffix ".yaml".
func removePlugin(plugin *Plugin) error {
	dir := filepath.Dir(plugin.BinPath)

	alias := Alias(plugin.info.Author, plugin.info.Name)
	if alias != "massa-labs/massa-wallet" {
		return os.RemoveAll(dir)
	}

	// for the massa-wallet we want to keep all the wallet entries
	entries, err := os.ReadDir(dir)
	if err != nil {
		return err
	}

	for _, entry := range entries {
		if !(strings.HasPrefix(entry.Name(), "wallet_") && strings.HasSuffix(entry.Name(), ".yaml")) {
			err = os.RemoveAll(filepath.Join(dir, entry.Name()))
			if err != nil {
				return err
			}
		}
	}

	return nil
}
```
The matching unit test:
```golang
func Test_removePlugin(t *testing.T) {
	testCases := []struct {
		name        string
		plugin      *Plugin
		createFiles []string
		remainFiles []string
	}{
		{
			name: "massa-labs/massa-wallet",
			plugin: &Plugin{
				info: &Information{
					Author: "massa-labs",
					Name:   "massa-wallet",
				},
			},
			createFiles: []string{"wallet_test.yaml", "other_file.txt"},
			remainFiles: []string{"wallet_test.yaml"},
		},
		{
			name: "massa-labs/not-massa-wallet",
			plugin: &Plugin{
				info: &Information{
					Author: "massa-labs",
					Name:   "massa-wallet",
				},
			},
			createFiles: []string{"wallet_test.yaml", "other_file.txt"},
			remainFiles: []string{},
		},
		{
			name: "not-massa-labs/massa-wallet",
			plugin: &Plugin{
				info: &Information{
					Author: "massa-labs",
					Name:   "massa-wallet",
				},
			},
			createFiles: []string{"wallet_test.yaml", "other_file.txt"},
			remainFiles: []string{},
		},
	}

	for _, tc := range testCases {
		t.Run(tc.name, func(t *testing.T) {
			// Create a temporary directory for testing.
			tempDir, err := ioutil.TempDir("", "plugin_test")
			if err != nil {
				t.Fatalf("Failed to create temp directory: %v", err)
			}

			tc.plugin.BinPath = filepath.Join(tempDir, "pluginBinary")

			// Create dummy files.
			for _, file := range tc.createFiles {
				dummyFile := filepath.Join(tempDir, file)
				if err := ioutil.WriteFile(dummyFile, []byte("test"), 0666); err != nil {
					t.Fatalf("Failed to create dummy file: %v", err)
				}
			}

			// Call the function to test.
			if err := removePlugin(tc.plugin); err != nil {
				t.Fatalf("Failed to remove plugin: %v", err)
			}

			// Check if the remaining files still exist.
			for _, file := range tc.remainFiles {
				dummyFile := filepath.Join(tempDir, file)
				if _, err := os.Stat(dummyFile); os.IsNotExist(err) {
					t.Fatalf("File was deleted: %v", err)
				}
			}

			// Clean up the temporary directory.
			os.RemoveAll(tempDir)
		})
	}
}
```

**Review by gregLibert — CHANGES_REQUESTED** [review]

**Review by gregLibert — APPROVED** [review]

**Comment by Thykof** [comment]
> * Tested on MacOS M1 - done.
> 
>     * Needs test for:
>       Linux
>       Windows

Great, it's the same behavior in all OS. No need to test on other OS if it work on one of them

**Comment by longgamma-massa** [comment]
@Thykof there might be a problem now. 

1. The wallet recognized my accounts when I tried to add them to the 'main' version. 
2. Now I downgraded to the latest-release version, the yaml files are saved on my computer, but there's no way to import them. I get 'filesystem error.
image

**Comment by longgamma-massa** [comment]
UPDATE: I think I found the issue (it was my old wallet.yaml) but would be great if you Q/A as well @Thykof

**Comment by Thykof** [comment]
I tested is works as expected:

- install the wallet plugin
- create or import an account
- remove the wallet plugin
- install it again
- see that your account is accessible from the plugin

**Pull request merged**